### ISSUE TITLE
False Cannot move on primitive types

### ISSUE THREAD

**Issue by krojew** [intellij-rust/intellij-rust#4858]
## Environment

* **IntelliJ Rust plugin version:** 0.2.114.2151-193
* **Rust toolchain version:** 1.42 nightly
* **IDE name and version:** CLion 2019.3.3
* **Operating system:** Windows 10

## Problem description

Since the last update, there are a ton of false positive "Cannot move" errors on copying primitive types:

```
fn get_time_multiplier(&self) -> f32 {
    self.time_multiplier // Cannot move
}
```

## Steps to reproduce

Create a struct with a filed of primitive type and a getter for it.

**Comment by Undin** [comment]
@ortem Could you take a look?

**Comment by ortem** [comment]
Cannot reproduce this.

@krojew do you have the experimental macro expansion engine enabled? If so, please try [disabling](https://www.jetbrains.com/help/clion/rust-support.html#project-settings) it and tell us if it helps.

**Comment by krojew** [comment]
I've invalidated caches and it seems to be ok now. I cannot reproduce the issue.

**Comment by ortem** [comment]
@krojew OK, that's great. Could you please check if you have the experimental macro expansion engine enabled, anyway? It will help us to find out the exact problem with caches.

**Comment by krojew** [comment]
Yes, I have it enabled.

**Comment by ortem** [comment]
@krojew Thank you! Use cache invalidation if you face the issue again, or try disabling the new macro engine if the problem occurs very often.

@vlad20012 the problem seems to be related to the new macro engine caches.

**Comment by Octogonapus** [comment]
I have also run into this issue.
Code to reproduce: https://pastebin.com/4HS4w9Se
The issue is on line 37, `self.ft * rhs.ft`: `self` and `rhs` are red-underlined with the tooltip "Cannot move".

* **IntelliJ Rust plugin version:** 0.2.115.2157-193
* **Rust toolchain version:** 1.41.0 (5e1a79984 2020-01-27) x86_64-unknown-linux-gnu
* **IDE name and version:** IntelliJ IDEA 2019.3.2 Ultimate Edition (IU-193.6015.39)
* **Operating system:** Linux 5.0.0-37-generic

**Comment by ariymarkowitz** [comment]
I am having this issue. The following is a minimal example:

```rust
struct Test {
    value: u32
}

impl Test {
    fn test(&self) -> u32 { self.value }
}
```

The error only appears when I include the dependency `bevy = "0.5"` (even though I am not using the package). I have not found any other packages that replicate this behaviour. Earlier versions of Bevy do not cause this behaviour.

I have tried invalidating the cache and restarting, and making a new project. The problem goes away when I set 'Expand declarative macros' to 'Use old engine'.
